lls/main.c

```diff
@@ -274,6 +274,67 @@ match_nd(struct rte_mbuf *pkt, struct gatekeeper_if *iface)
 	return 0;
 }
 
+/*
+ * Adjust the L2 header to be the correct size when sent out of @iface.
+ *
+ * This function is called on a @pkt that was previously received from
+ * another interface, and therefore may not have the same L2 space as
+ * that which @iface requires.
+ *
+ * This function assumes that the Ethernet header has already been
+ * filled-in, so if the L2 space is changed it shifts around the
+ * Ethernet header and applies the VLAN tag as needed.
+ */
+int
+adjust_shift_l2_hdr(struct gatekeeper_if *iface, struct rte_mbuf *pkt,
```

Comment:
I'm under the impression that `adjust_shift_l2_hdr()` could reuse `adjust_pkt_len()`, or a more general version of `adjust_pkt_len()`. They share a lot of code.